relay/network_graph/dijkstra_weighted.py

```diff
@@ -111,7 +46,81 @@ def find_path(G, source, target, get_fee, value, max_hops=None, max_fees=None):
                 push(fringe, (n+1, vu_dist, next(c), u))
                 paths[u] = paths[v] + [u]
     try:
-        return (dist[target]-value, paths[target])
+        return (dist[target]-value, paths[target])  # cost is the total fee, not the actual amount to be transfered
     except KeyError:
         raise nx.NetworkXNoPath(
             "node %s not reachable from %s" % (source, target))
+
+
+def find_path_triangulation(G, source, target_reduce, target_increase, get_fee, value, max_hops=None, max_fees=None):
+    # target_reduce is the node we want to reduce our credit with (for value > 0)
+    # target_increase is the node which will result with an increased debt (for value > 0)
+    # must return a path and the total fee of that path
+    # to be called in the right order with source as source and target as target, contrary to find_path
+
+    # value may be negative as far as I know
+    # value < 0 means source wants to increase the amount target_reduce owes her
+        # (give negative value to target_reduce)
+
+    G_succ = G.succ if G.is_directed() else G.adj
+
+    neighbors = [x[0] for x in G_succ[source].items()]
+
+    if target_reduce not in neighbors:
+        raise nx.NetworkXNoPath(
+            "node %s not a neighbor of %s" % (target_reduce, source))
+
+    if target_increase not in neighbors:
+        raise nx.NetworkXNoPath(
+            "node %s not a neighbor of %s" % (target_increase, source))
+
+    if value >= 0:
+        if source < target_reduce:
```

Comment:
Please fix the get_fee function, that we dont have to do that every time check every time before calling the function.

Reply:
Or you can write a wrapper function that does that.

Reply:
If you feel like this function is not need outside you can also define it as a inner function. https://realpython.com/inner-functions-what-are-they-good-for/